### PR TITLE
feat(pattern/select): create select form item

### PR DESCRIPTION
**Pull request by gael-boyenval**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/adeo/design-system--styleguide/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

## Other information

### CODE REVIEW

File: src/pages/Components/Select/index.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+<hintitem>
+  It's recommended to use shorts option texts.
+</hintitem>
+
+<hintitem>
+ Always add ‘-- ’ and ‘ --’ in before and after a first option used as a placeholder and Capitalise the first letter of a select option.
+</hintitem>
+
+```html
```
Comment (caqueste) [inline src/pages/Components/Select/index.md]: I think for a better understanding, we can put a preview inside a hint box.

( <preview path="src/pages/Components/Select/previews/Select-default"></preview> )

File: src/pages/Components/Select/index.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+<hintitem>
+  Always use a select maximum width in relation to the options line length.
+</hintitem>
+
+<hintitem>
+  It's recommended to use shorts option texts.
+</hintitem>
+
+<hintitem>
+ Always add ‘-- ’ and ‘ --’ in before and after a first option used as a placeholder and Capitalise the first letter of a select option.
```
Comment (caqueste) [inline src/pages/Components/Select/index.md]: Maybe you can split this do section in two different ones.
For example : 
*  Always use ‘—‘ before and after the first option used as a placeholder 
* Capitalise the first letter of a select option

Comment (gael-boyenval) [reply src/pages/Components/Select/index.md]: "With spaces between the dashes and the text". I agree with the splitting too

File: src/pages/Components/Select/index.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+</hintitem>
+
+```html
+<option>-- Select an option --</option>
+```
+<hintitem dont="true">
+  Don't use select with very short list of options (for example, 7 values), use radio buttons instead.
```
Comment (gael-boyenval) [inline src/pages/Components/Select/index.md]: too blurry, I think we should use a min and a max, and not say "for example" witch does not give any rule to follow.

File: src/pages/Components/Select/index.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+</hintitem>
+
+<hintitem dont="true">
+  Don't use select with infinite options (for example, more than 15 values), they are hard to scan. It's recommended to use an autocomplete search field inside a dropdown.
```
Comment (gael-boyenval) [inline src/pages/Components/Select/index.md]: infinite ? WTF ? same here I really think we should have a min and a max.

File: src/pages/Components/Select/code.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+- `focus`
+- `disabled`
+- `is-valid`
+- `is-invalid`
+
+
+<preview path="src/pages/Components/Select/previews/Select-all-states"></preview>
+
+## Default option (as a placeholder)
+ By default, the first option is displayed. You can use it as a placeholder, but remember to distinguish it from the rest of options and to add an empty value attribute.
```
Comment (gael-boyenval) [inline src/pages/Components/Select/code.md]: I think we can be more precise and explain the `-- Label --` pattern again.

File: src/pages/Components/Select/index.md (hunk 4 above)
Comment (gael-boyenval) [reply src/pages/Components/Select/index.md]: 15 is not infinite

File: src/pages/Components/Select/index.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+ Always Capitalise the first letter of a select option.
+</hintitem>
+
+<hint type="do" title="Always use '--' before and after the first option used as a placeholder with spaces between the dashes and the text.">
+    <preview path="src/pages/Components/Select/previews/Select-default" nude=true ></preview>
+</hint>
+
+
+<hintitem dont="true">
+  Don't use select with very short list of options, if you have more than 7 options, use radio buttons instead.
```
Comment (gael-boyenval) [inline src/pages/Components/Select/index.md]: less than

**Pull request merged**